File: JS/01.Basics/99.Pre-Exam/05.Everest.js
function mountainClimbing(inputArray) {
    let height = 5364;
    const top = 8848;
    let inputIndex = 0;
    let inputString = inputArray[inputIndex++];
    let up = parseInt(inputArray[inputIndex++]);
    let days = 1;

    while (true) {
        if (inputString !== "END") {
            height += up;
        }

        if (inputString === "Yes") {
            days++;
        }

        if (inputString === "END" || days >= 5 || height >= top) {
            break;
        }

        inputString = inputArray[inputIndex++];

        if (inputString !== "END") {
            up = parseInt(inputArray[inputIndex++]);
        }
    }

    if (top > height) {
        console.log("Failed!");
        console.log(height);
    } else {
        console.log(`Goal reached for ${days} days!`);
    }
}

// Example usage:
const inputArray = ["Yes", "100", "No", "50", "Yes", "200", "END"];
mountainClimbing(inputArray);

File: Python/01.ProgrammingBasics/99.Pre-Exam/05.Everest/_05.Everest.py
height = 5364
top = 8848
input_string = input()
up = int(input())
days = 1

while True:
    if input_string != "END":
        height += up

    if input_string == "Yes":
        days += 1

    if input_string == "END" or days >= 5 or height >= top:
        break

    input_string = input()

    if input_string != "END":
        up = int(input())

if top > height:
    print("Failed!")
    print(f"{height}")
else:
    print(f"Goal reached for {days} days!")
